ldap/servers/slapd/pagedresults.c

```diff
@@ -924,19 +931,25 @@ pagedresults_is_timedout_nolock(Connection *conn)
 
     slapi_log_err(SLAPI_LOG_TRACE, "pagedresults_is_timedout", "=>\n");
 
+    pthread_mutex_lock(pageresult_lock_get_addr(conn));
```

Comment:
I have concern with a comment in pageresult_lock_get_addr. it is said to not lock pageresult_lock while holding c_mutex. IIUC here we are holding c_mutex.

Reply:
This is required to fix the problem :-)  Other places in the code (pagedresult.c) do the exact same thing

Reply:
Actually I'm not sure if other places do that, but this is the fix that worked for the customer.  I'll double check that...

Reply:
Well we also have a comment say we can not call the the timedout function unless that lock is taken :-)

```
 *
 * must be called within pageresult_lock_get_addr(conn)
 */
int
pagedresults_is_timedout_nolock(Connection *conn)
```

Reply:
I am afraid that @tbordaz is right: this change could lead to a deadlock:
       setup_pr_read_pds held the c_mutex when calling the pagedresults_is_timedout_nolock
IMHO that is not the case for the other 'exact same thing'

Reply:
But setup_pr_read_pds() **always** takes the conn lock before calling pagedresults_is_timedout_nolock, so why aren't we seeing a deadlock then?

Reply:
But setup_pr_read_pds() always takes the conn lock before calling pagedresults_is_timedout_nolock, so why aren't we seeing a deadlock then?

Because
 1 - before your change pagedresults_is_timedout_nolock was not taking the pageresult_lock_get_addr so no race condition
  2 - with your code there is a race condition but there must have other thread that hold both lock at the right time and the critical section is very small so the probability is very low